src/renderer/store/certificates/certificates.saga.ts

```diff
@@ -0,0 +1,50 @@
+import { all, takeEvery, put, apply } from 'redux-saga/effects'
+import { call } from 'typed-redux-saga'
+
+import { certificatesActions } from './certificates.reducer'
+import { createUserCsr } from '../../pkijs/generatePems/requestCertificate'
+import { createUserCert } from '../../pkijs/generatePems/generateUserCertificate'
+import { dataFromRootPems } from '../../../shared/static'
+import electronStore from '../../../shared/electronStore'
+import { PayloadAction } from '@reduxjs/toolkit'
+
+export function* responseGetCertificates(
+  action: PayloadAction<ReturnType<typeof certificatesActions.responseGetCertificates>['payload']>
+): Generator {
+  const certificates = action.payload
+  yield put(certificatesActions.setUsersCertificates(certificates))
+}
+
+const getDate = (date?: string) => new Date(date)
+
+export function* creactOwnCertificate(
+  action: PayloadAction<ReturnType<typeof certificatesActions.creactOwnCertificate>['payload']>
+): Generator {
+  const certString = dataFromRootPems.certificate
+  const keyString = dataFromRootPems.privKey
+
+  const notBeforeDate = yield call(getDate)
+  const notAfterDate = yield call(getDate, '1/1/2031')
+
+  yield apply(electronStore, electronStore.get, ['hiddenServices'])
```

Comment:
Thos is to get onion address few lines below.

You're still calliny there electroStore.get without yieldong effect